emu/src/io_registers.rs

```diff
@@ -0,0 +1,342 @@
+use crate::bitwise::Bits;
+
+pub enum IORegisterAccessControl {
+    Read,
+    Write,
+    ReadWrite,
+}
+
+pub struct IORegister {
+    value: u32,
+    access: IORegisterAccessControl,
+}
+
+impl IORegister {
+    pub fn read(&self) -> u32 {
+        match self.access {
+            IORegisterAccessControl::Read | IORegisterAccessControl::ReadWrite => self.value,
+            _ => panic!("Requested IO Register cannot be read."),
+        }
+    }
+
+    pub fn write(&mut self, value: u32) {
+        match self.access {
+            IORegisterAccessControl::Write | IORegisterAccessControl::ReadWrite => {
+                self.value = value;
+            }
+
+            _ => panic!("Requested IO Register cannot be written."),
+        }
+    }
+
+    pub fn set_byte(&mut self, byte: u8, value: u8) {
+        match self.access {
+            IORegisterAccessControl::Write | IORegisterAccessControl::ReadWrite => {
+                self.value.set_byte(byte, value);
+            }
+
+            _ => panic!("Requested IO Register cannot be written."),
+        }
+    }
+}
+
+#[allow(non_snake_case)]
+pub struct LCDRegisters {
+    // LCD Control
```

Comment:
This kind of comment are just comment, if you want to make as a doc you should use `///`

Reply:
I didn't know that, thanks!